### PR TITLE
ASoC: SOF: remove superfluous initialization

### PR DESCRIPTION
**Pull request by ranj063**
remove superfluous initialization everywhere based on Takashi's
comments

Not sure if I overdid this. Can I request for this to be checked?

### CODE REVIEW

File: sound/soc/sof/hw-spi.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ static int spi_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 static int spi_get_reply(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 {
 	struct sof_ipc_reply reply;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/hw-spi.c]: It seems ret could be uninitialized in this function.

File: sound/soc/sof/intel/bdw.c [hunk 2]
```diff
@@ -486,7 +486,7 @@ static void bdw_get_reply(struct snd_sof_dev *sdev)
 	struct snd_sof_ipc_msg *msg = sdev->msg;
 	struct sof_ipc_reply reply;
 	unsigned long flags;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/intel/bdw.c]: same as here

File: sound/soc/sof/intel/byt.c [hunk 3]
```diff
@@ -384,7 +384,7 @@ static void byt_get_reply(struct snd_sof_dev *sdev)
 	struct snd_sof_ipc_msg *msg = sdev->msg;
 	struct sof_ipc_reply reply;
 	unsigned long flags;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/intel/byt.c]: same here

File: sound/soc/sof/intel/hda-ctrl.c [hunk 4]
```diff
@@ -56,7 +56,7 @@ int hda_dsp_ctrl_get_caps(struct snd_sof_dev *sdev)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	u32 cap, offset, feature;
-	int count = 0;
+	int count;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda-ctrl.c]: same here

File: sound/soc/sof/intel/hda-ipc.c [hunk 5]
```diff
@@ -73,7 +73,7 @@ void hda_dsp_ipc_get_reply(struct snd_sof_dev *sdev)
 	struct sof_ipc_reply reply;
 	struct sof_ipc_cmd_hdr *hdr;
 	unsigned long flags;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda-ipc.c]: same here

File: sound/soc/sof/intel/hda-stream.c [hunk 6]
```diff
@@ -131,7 +131,7 @@ int hda_dsp_stream_spib_config(struct snd_sof_dev *sdev,
 			       int enable, u32 size)
 {
 	struct hdac_stream *hstream = &stream->hstream;
-	u32 mask = 0;
+	u32 mask;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda-stream.c:134]: initial here or modify line 141 to ``mask = (1 << hstream->index);``

File: sound/soc/sof/intel/hda-stream.c [hunk 7]
```diff
@@ -156,7 +156,7 @@ struct hdac_ext_stream *
 hda_dsp_stream_get(struct snd_sof_dev *sdev, int direction)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
-	struct hdac_ext_stream *stream = NULL;
+	struct hdac_ext_stream *stream;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda-stream.c]: same here

File: sound/soc/sof/intel/hda.c [hunk 8]
```diff
@@ -183,7 +183,7 @@ static int hda_init(struct snd_sof_dev *sdev)
 {
 	struct hda_bus *hbus;
 	struct hdac_bus *bus;
-	struct hdac_ext_bus_ops *ext_ops = NULL;
+	struct hdac_ext_bus_ops *ext_ops;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda.c]: If CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC is not enabled, *ext_ops will not be initialized

File: sound/soc/sof/intel/hda.c [hunk 9]
```diff
@@ -227,7 +227,7 @@ static int hda_init(struct snd_sof_dev *sdev)
 static const char *fixup_tplg_name(struct snd_sof_dev *sdev,
 				   const char *sof_tplg_filename)
 {
-	const char *tplg_filename = NULL;
+	const char *tplg_filename;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda.c]: same here

File: sound/soc/sof/intel/hsw.c [hunk 10]
```diff
@@ -487,7 +487,7 @@ static void hsw_get_reply(struct snd_sof_dev *sdev)
 	struct snd_sof_ipc_msg *msg = sdev->msg;
 	struct sof_ipc_reply reply;
 	unsigned long flags;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/intel/hsw.c]: same here

File: sound/soc/sof/ipc.c [hunk 11]
```diff
@@ -56,7 +56,7 @@ struct sof_ipc_ctrl_data_params {
 static void ipc_log_header(struct device *dev, u8 *text, u32 cmd)
 {
 	u8 *str;
-	u8 *str2 = NULL;
+	u8 *str2;
```
Comment (bardliao) [inline sound/soc/sof/ipc.c]: same here

File: sound/soc/sof/ipc.c [hunk 12]
```diff
@@ -598,10 +598,10 @@ static int sof_set_get_large_ctrl_data(struct snd_sof_dev *sdev,
 {
 	struct sof_ipc_ctrl_data *partdata;
 	size_t send_bytes;
-	size_t offset = 0;
+	size_t offset;
```
Comment (bardliao) [inline sound/soc/sof/ipc.c]: same here

File: sound/soc/sof/loader.c [hunk 13]
```diff
@@ -38,7 +38,7 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 {
 	struct sof_ipc_ext_data_hdr *ext_hdr;
 	void *ext_data;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/loader.c]: same here

File: sound/soc/sof/pcm.c [hunk 14]
```diff
@@ -528,7 +528,7 @@ static int sof_pcm_new(struct snd_soc_pcm_runtime *rtd)
 	struct snd_sof_pcm *spcm;
 	struct snd_pcm *pcm = rtd->pcm;
 	struct snd_soc_tplg_stream_caps *caps;
-	int ret = 0, stream = SNDRV_PCM_STREAM_PLAYBACK;
+	int ret, stream = SNDRV_PCM_STREAM_PLAYBACK;
```
Comment (bardliao) [inline sound/soc/sof/pcm.c]: It seems we can remove ret and return 0. I didn't see any case ret will be non zero.

File: sound/soc/sof/pm.c [hunk 15]
```diff
@@ -13,9 +13,9 @@
 
 static int sof_restore_kcontrols(struct snd_sof_dev *sdev)
 {
-	struct snd_sof_control *scontrol = NULL;
+	struct snd_sof_control *scontrol;
 	int ipc_cmd, ctrl_type;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/pm.c]: same here

File: sound/soc/sof/topology.c [hunk 16]
```diff
@@ -727,7 +727,7 @@ static void sof_parse_word_tokens(struct snd_soc_component *scomp,
 	size_t size = sizeof(struct sof_ipc_dai_dmic_pdm_ctrl);
 	int i, j;
 	u32 offset;
-	u32 *index = NULL;
+	u32 *index;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/hw-spi.c (hunk 1 above)
Comment (lyakh) [inline sound/soc/sof/hw-spi.c]: This is wrong, in the else-else branch below (cannot comment there...) at line 198, ret remains unassigned and then it is returned.

File: sound/soc/sof/topology.c [hunk 17]
```diff
@@ -1565,13 +1565,13 @@ static int sof_effect_fir_load(struct snd_soc_component *scomp, int index,
 {
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct snd_soc_tplg_private *private = &tw->priv;
-	struct snd_sof_control *scontrol = NULL;
+	struct snd_sof_control *scontrol;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/topology.c [hunk 18]
```diff
@@ -1915,7 +1915,7 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 	struct soc_bytes_ext *sbe;
 	struct snd_sof_dai *dai;
 	struct soc_enum *se;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/topology.c [hunk 19]
```diff
@@ -2139,7 +2139,7 @@ static int sof_set_dai_config(struct snd_sof_dev *sdev, u32 size,
 			      struct sof_ipc_dai_config *config)
 {
 	struct snd_sof_dai *dai;
-	int found = 0;
+	int found;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/topology.c [hunk 20]
```diff
@@ -2385,7 +2385,7 @@ static int sof_link_hda_process(struct snd_sof_dev *sdev,
 	struct sof_ipc_reply reply;
 	u32 size = sizeof(*config);
 	struct snd_sof_dai *sof_dai;
-	int found = 0;
+	int found;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/topology.c [hunk 21]
```diff
@@ -2453,10 +2453,10 @@ static int sof_link_hda_load(struct snd_soc_component *scomp, int index,
 	struct snd_soc_tplg_private *private = &cfg->priv;
 	struct snd_soc_dai *dai;
 	u32 size = sizeof(*config);
-	u32 tx_num = 0;
-	u32 tx_slot = 0;
-	u32 rx_num = 0;
-	u32 rx_slot = 0;
+	u32 tx_num;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/intel/bdw.c [hunk 22]
```diff
@@ -326,12 +326,12 @@ static irqreturn_t bdw_irq_thread(int irq, void *context)
...
-	u32 inbox_offset = 0;
-	u32 outbox_size = 0;
-	u32 stream_size = 0;
-	u32 inbox_size = 0;
+	u32 outbox_offset;
+	u32 stream_offset;
+	u32 inbox_offset;
+	u32 outbox_size;
+	u32 stream_size;
+	u32 inbox_size;
```
Comment (lyakh) [inline sound/soc/sof/intel/bdw.c]: this looks wrong to me, unless we're guaranteed that the loop below will definitely first run through all the required cases to initialise all those variables. At least formally, when there isn't such a guarantee, e.g. if num_windows == 1, the loop is run only once, so at least one of inbox_size or outbox_size will stay uninitialised.

File: sound/soc/sof/intel/bdw.c (hunk 2 above)
Comment (lyakh) [inline sound/soc/sof/intel/bdw.c]: Wrong too, else-else leaves "ret" uninitialised

File: sound/soc/sof/topology.c [hunk 23]
```diff
@@ -2613,7 +2613,7 @@ static int sof_link_hda_unload(struct snd_sof_dev *sdev,
 {
 	struct snd_soc_dai_link_component dai_component;
 	struct snd_soc_dai *dai;
-	int ret = 0;
+	int ret;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/intel/byt.c [hunk 24]
```diff
@@ -115,12 +115,12 @@ static void byt_get_reply(struct snd_sof_dev *sdev);
...
-	u32 inbox_offset = 0;
-	u32 outbox_size = 0;
-	u32 stream_size = 0;
-	u32 inbox_size = 0;
+	u32 outbox_offset;
+	u32 stream_offset;
+	u32 inbox_offset;
+	u32 outbox_size;
+	u32 stream_size;
+	u32 inbox_size;
```
Comment (lyakh) [inline sound/soc/sof/intel/byt.c]: looks wrong - see above

File: sound/soc/sof/topology.c [hunk 25]
```diff
@@ -2645,8 +2645,8 @@ static int sof_link_unload(struct snd_soc_component *scomp,
 	struct snd_soc_dai_link *link =
 		container_of(dobj, struct snd_soc_dai_link, dobj);
 
-	struct snd_sof_dai *sof_dai = NULL;
-	int ret = 0;
```
Comment (bardliao) [inline sound/soc/sof/topology.c]: same here

File: sound/soc/sof/intel/byt.c (hunk 3 above)
Comment (lyakh) [inline sound/soc/sof/intel/byt.c]: similar to an earlier snippet - looks wrong to me.

File: sound/soc/sof/intel/hda-ctrl.c [hunk 26]
```diff
@@ -27,7 +27,7 @@
 int hda_dsp_ctrl_link_reset(struct snd_sof_dev *sdev, bool reset)
 {
 	unsigned long timeout;
-	u32 gctl = 0;
+	u32 gctl;
```
Comment (lyakh) [inline sound/soc/sof/intel/hda-ctrl.c]: theoretically this is wrong too - if the loop below isn't entered, gctl remains uninitialised and it's then printed out

File: sound/soc/sof/intel/hda-ctrl.c (hunk 4 above)
Comment (lyakh) [inline sound/soc/sof/intel/hda-ctrl.c]: wrong. to use count++ you need to initialise it first

File: sound/soc/sof/intel/hda-ipc.c (hunk 5 above)
Comment (lyakh) [inline sound/soc/sof/intel/hda-ipc.c]: wrong again: else-else

File: sound/soc/sof/intel/hda-ipc.c [hunk 27]
```diff
@@ -248,12 +248,12 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
...
-	u32 inbox_offset = 0;
-	u32 outbox_size = 0;
-	u32 stream_size = 0;
-	u32 inbox_size = 0;
+	u32 outbox_offset;
+	u32 stream_offset;
+	u32 inbox_offset;
+	u32 outbox_size;
+	u32 stream_size;
+	u32 inbox_size;
```
Comment (lyakh) [inline sound/soc/sof/intel/hda-ipc.c]: see above

File: sound/soc/sof/intel/hda-loader.c [hunk 28]
```diff
@@ -27,7 +27,7 @@ static int cl_stream_prepare(struct snd_sof_dev *sdev, unsigned int format,
 			     unsigned int size, struct snd_dma_buffer *dmab,
 			     int direction)
 {
-	struct hdac_ext_stream *dsp_stream = NULL;
+	struct hdac_ext_stream *dsp_stream;
```
Comment (lyakh) [inline sound/soc/sof/intel/hda-loader.c:30]: to make this even more obvious you could modify the if below to:
```
if (direction != SNDRV_PCM_STREAM_PLAYBACK) {
	dev_err(sdev->dev, "error: code loading DMA is playback only\n");
	return -EINVAL;
}
dsp_stream = hda_dsp_stream_get(sdev, direction);
```
then dsp_stream initialisation will become even more obvious.

File: sound/soc/sof/intel/hda-stream.c (hunk 6 above)
Comment (lyakh) [inline sound/soc/sof/intel/hda-stream.c:134]: wrong, but to make this correct you have to modify the line below to
```
mask = (1 << hstream->index);
```

File: sound/soc/sof/intel/hda-stream.c (hunk 7 above)
Comment (lyakh) [inline sound/soc/sof/intel/hda-stream.c]: wrong

File: sound/soc/sof/intel/hda.c (hunk 8 above)
Comment (lyakh) [inline sound/soc/sof/intel/hda.c]: wrong for CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC not enabled

File: sound/soc/sof/intel/hda.c (hunk 9 above)
Comment (lyakh) [inline sound/soc/sof/intel/hda.c]: wrong if (!split_ext)

File: sound/soc/sof/intel/hsw.c [hunk 29]
```diff
@@ -327,12 +327,12 @@ static irqreturn_t hsw_irq_thread(int irq, void *context)
...
-	u32 inbox_offset = 0;
-	u32 outbox_size = 0;
-	u32 stream_size = 0;
-	u32 inbox_size = 0;
+	u32 outbox_offset;
+	u32 stream_offset;
+	u32 inbox_offset;
+	u32 outbox_size;
+	u32 stream_size;
+	u32 inbox_size;
```
Comment (lyakh) [inline sound/soc/sof/intel/hsw.c]: wrong as above

File: sound/soc/sof/intel/hsw.c (hunk 10 above)
Comment (lyakh) [inline sound/soc/sof/intel/hsw.c]: wrong in else-else

File: sound/soc/sof/ipc.c (hunk 11 above)
Comment (lyakh) [inline sound/soc/sof/ipc.c]: wrong

File: sound/soc/sof/ipc.c [hunk 30]
```diff
@@ -598,10 +598,10 @@ static int sof_set_get_large_ctrl_data(struct snd_sof_dev *sdev,
...
 	size_t msg_bytes;
 	size_t pl_size;
-	int err = 0;
+	int err;
```
Comment (lyakh) [inline sound/soc/sof/ipc.c]: both wrong

File: sound/soc/sof/loader.c (hunk 13 above)
Comment (lyakh) [inline sound/soc/sof/loader.c]: wrong

File: sound/soc/sof/pcm.c (hunk 14 above)
Comment (lyakh) [inline sound/soc/sof/pcm.c]: wrong as is, but please remove "ret" completely in this function and convert below to "return 0"

File: sound/soc/sof/pm.c (hunk 15 above)
Comment (lyakh) [inline sound/soc/sof/pm.c]: scontrol correct, ret wrong

File: sound/soc/sof/topology.c (hunk 16 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

File: sound/soc/sof/topology.c (hunk 17 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

File: sound/soc/sof/topology.c [hunk 31]
```diff
@@ -1565,13 +1565,13 @@ static int sof_effect_fir_load(struct snd_soc_component *scomp, int index,
 {
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct snd_soc_tplg_private *private = &tw->priv;
-	struct snd_sof_control *scontrol = NULL;
+	struct snd_sof_control *scontrol;
 	struct snd_soc_dapm_widget *widget = swidget->widget;
-	const struct snd_kcontrol_new *kc = NULL;
+	const struct snd_kcontrol_new *kc;
```
Comment (lyakh) [inline sound/soc/sof/topology.c:1730]: correct

File: sound/soc/sof/topology.c [hunk 32]
```diff
@@ -1565,13 +1565,13 @@ static int sof_effect_fir_load(struct snd_soc_component *scomp, int index,
...
 	struct soc_bytes_ext *sbe;
-	struct sof_abi_hdr *pdata = NULL;
+	struct sof_abi_hdr *pdata;
 	struct sof_ipc_comp_eq_fir *fir;
-	size_t ipc_size = 0, fir_data_size = 0;
+	size_t ipc_size, fir_data_size;
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: ipc_size correct, others wrong

File: sound/soc/sof/topology.c [hunk 33]
```diff
@@ -1644,12 +1644,12 @@ static int sof_effect_iir_load(struct snd_soc_component *scomp, int index,
...
-	const struct snd_kcontrol_new *kc = NULL;
+	const struct snd_kcontrol_new *kc;
 	struct soc_bytes_ext *sbe;
-	struct snd_sof_control *scontrol = NULL;
-	struct sof_abi_hdr *pdata = NULL;
+	struct snd_sof_control *scontrol;
+	struct sof_abi_hdr *pdata;
 	struct sof_ipc_comp_eq_iir *iir;
-	size_t ipc_size = 0, iir_data_size = 0;
+	size_t ipc_size, iir_data_size;
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: see above

File: sound/soc/sof/topology.c [hunk 34]
```diff
@@ -1778,8 +1778,8 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 	struct snd_sof_widget *swidget;
 	struct snd_sof_dai *dai;
 	struct sof_ipc_comp_reply reply;
-	struct snd_sof_control *scontrol = NULL;
-	int ret = 0;
+	struct snd_sof_control *scontrol;
+	int ret;
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: ret is wrong, scontrol correct

File: sound/soc/sof/topology.c (hunk 18 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

File: sound/soc/sof/topology.c [hunk 35]
```diff
@@ -2001,7 +2001,7 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 	struct snd_soc_tplg_stream_caps *caps;
 	struct snd_sof_pcm *spcm;
 	int stream = SNDRV_PCM_STREAM_PLAYBACK;
-	int ret = 0;
+	int ret;
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong because of goto capture

File: sound/soc/sof/topology.c (hunk 19 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

File: sound/soc/sof/topology.c (hunk 20 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

File: sound/soc/sof/topology.c [hunk 36]
```diff
@@ -2453,10 +2453,10 @@ static int sof_link_hda_load(struct snd_soc_component *scomp, int index,
...
 	struct snd_soc_dai *dai;
 	u32 size = sizeof(*config);
-	u32 tx_num = 0;
-	u32 tx_slot = 0;
-	u32 rx_num = 0;
-	u32 rx_slot = 0;
+	u32 tx_num;
+	u32 tx_slot;
+	u32 rx_num;
+	u32 rx_slot;
```
Comment (lyakh) [inline sound/soc/sof/topology.c]: potentially all wrong

File: sound/soc/sof/topology.c (hunk 23 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

File: sound/soc/sof/topology.c [hunk 37]
```diff
@@ -2645,8 +2645,8 @@ static int sof_link_unload(struct snd_soc_component *scomp,
 	struct snd_soc_dai_link *link =
 		container_of(dobj, struct snd_soc_dai_link, dobj);
 
-	struct snd_sof_dai *sof_dai = NULL;
```
Comment (lyakh) [inline sound/soc/sof/topology.c:2837]: correct

File: sound/soc/sof/topology.c (hunk 25 above)
Comment (lyakh) [inline sound/soc/sof/topology.c]: wrong

**Comment by ranj063** [comment]
Thanks @lyakh and @bardliao I'll fix these up today.

**Comment by ranj063** [comment]
@lyakh @bardliao updated based on your comments now.

**Comment by plbossart** [comment]
@ranj063 looks good, but the topology parts need to be updated now that we use the process and not effects. Can you rebase and resubmit?

**Comment by ranj063** [comment]
> @ranj063 looks good, but the topology parts need to be updated now that we use the process and not effects. Can you rebase and resubmit?

@plbossart done

**Comment by plbossart** [comment]
> > @ranj063 looks good, but the topology parts need to be updated now that we use the process and not effects. Can you rebase and resubmit?
> 
> @plbossart done

failed build?

**Comment by ranj063** [comment]
hmm there're a couple of sparse issue. Let me check today.

**Comment by ranj063** [comment]
@plbossart all tests passed now.

**Pull request merged**